go/vt/vtgate/planbuilder/jointab.go

```diff
@@ -67,6 +68,33 @@ func (jt *jointab) Procure(bldr builder, col *sqlparser.ColName, to int) string
 	return joinVar
 }
 
+// GenerateSubqueryVars generates substitution variable names for
+// a subquery. It returns three names based on: __sq, __has_values,
+// __is_empty. The appropriate names can be used for substitution
+// depending on the scenario.
+func (jt *jointab) GenerateSubqueryVars() (sq, hasValues string) {
+	for {
+		jt.varIndex++
+		suffix := strconv.Itoa(jt.varIndex)
+		var1 := "__sq" + suffix
+		var2 := "__has_values" + suffix
```

Comment:
Something which just occurred to me -- we should probably be more consistent in the naming scheme for these bind vars. Sorry for suggesting yet another renaming change, but I think it's worth it for clarity.

Specifically it seems like it'd be more consistent to change these to "__sq_valueX" and "__sq_has_valuesX" (or maybe invert the boolean sense of the latter and use "__sq_valueX" and "__sq_emptyX")?

Reply:
I went with `__sq_has_valuesX`. The 'is_empty' approach felt less intuitive for the EXISTS case.